src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -142,28 +141,35 @@ export module DotNet {
   /**
    * Creates a JavaScript data reference that can be passed to .NET via interop calls.
    *
-   * @param arrayBufferView The ArrayBufferView used to create the JavaScript stream reference.
+   * @param streamReference The ArrayBufferView or Blob used to create the JavaScript stream reference.
    * @returns The JavaScript data reference (this will be the same instance as the given object).
    * @throws Error if the given value is not an Object or doesn't have a valid byteLength.
    */
-  export function createJSStreamReference(arrayBufferView: ArrayBufferView | any): any {
+  export function createJSStreamReference(streamReference: ArrayBufferView | Blob | any): any {
+    let length = -1;
     // Check if this is an ArrayBufferView, and if it has a valid byteLength for transfer
     // using a JSStreamReference.
-    if (!(arrayBufferView.buffer instanceof ArrayBuffer)) {
-      throw new Error(`Cannot create a JSStreamReference from the value '${arrayBufferView}' as it is not have a 'buffer' property of type 'ArrayBuffer'.`);
-    } else if (arrayBufferView.byteLength === undefined) {
-      throw new Error(`Cannot create a JSStreamReference from the value '${arrayBufferView}' as it doesn't have a byteLength.`);
+    if (streamReference instanceof Blob) {
+      length = streamReference.size;
+    } else if (streamReference.buffer instanceof ArrayBuffer) {
+      if (streamReference.byteLength === undefined) {
```

Comment:
Not sure about this check. Since we know it's `instanceof ArrayBuffer` whose [prototype declares this property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/byteLength), it's always going to be true, isn't it?

I don't think we have to worry about cases where someone deliberately overwrites standard properties on objects with bad values - we can just let it crash or fail in ways it would for a malicious client.

Or is there some scenario where `byteLength` can be `undefined` under normal circumstances?

Reply:
Update: sorry, I misread the code. I see that you're checking `byteLength` on a different object than `.buffer`.